pyrometheus/__init__.py

```diff
@@ -216,8 +225,111 @@ def equilibrium_constants_expr(sol: ct.Solution, react: ct.Reaction, gibbs_rt):
 
 # {{{ Rate coefficients
 
-def arrhenius_expr(rate_coeff: ct.Arrhenius):
-    raise NotImplementedError()
+def rate_coefficient_expr(rate_coeff: ct.Arrhenius, t):
+    """:returns: The rate coefficient expression for *rate_coeff* in terms
+    of the temperature *t* as a :class:`pymbolic.primitives.Expression`"""
+    # Rate parameters
+    a = rate_coeff.pre_exponential_factor
+    b = rate_coeff.temperature_exponent
+    t_a = rate_coeff.activation_energy/ct.gas_constant
+    if b == 0 and t_a == 0:
+        # Constant rate
+        return a
+    elif b != 0 and t_a == 0:
+        # Weakly temperature-dependent rate
+        return a * t**b
+    elif b == 0 and a != 0 and t_a != 0:
+        # Classic Arrhenius rate
+        return p.Variable("exp")(np.log(a)-t_a/t)
+    else:
+        # Modified Arrhenius
+        return p.Variable("exp")(np.log(a)+b*p.Variable("log")(t)-t_a/t)
+
+
+def third_body_efficiencies_expr(sol: ct.Solution, react: ct.Reaction, c):
+    """:returns: The third-body concentration expression for reaction *react* in terms
+    of the species concentraions *c* as a :class:`pymbolic.primitives.Expression`"""
```

Comment:
What data type is *c*?

Reply:
""":returns: A :class:`pymbolic.primitives.Expression` for the third-body concentration of reaction *react* in terms of a :class:`pymbolic.primitives.Expression` for the species concentrations *c*"""
?

Reply:
This one I'm not so sure about.  The species concentrations is a DOFArray in MIRGE-Com, and an np.array otherwise.  I wonder how we should document its type.  I don't think it would be right to mention DOFArray down inside Pyro.  Maybe we need to document `usr_np array type` or so... and then refer to it?  Just not sure.  I think this is something we can discuss - and (at least from my perspective) should not prevent merging.